packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx

```diff
@@ -1,427 +1,187 @@
-            styles.searchIconWrapper,
-            styles.searchIconClearWrapper
-          ) }>
-            <Icon iconName='cancel' />
-          </div>
-        </div>
-      );
-    }
+    const {
+      className,
+      items,
+      overflowItems,
+      farItems,
+      elipisisAriaLabel,
+      elipisisIconProps,
+      buttonStyles,
+      onRenderButton = this._onRenderButton,
+      onReduceData = this._onReduceData,
+      onGrowData = this._onGrowData,
+    } = this.props;
+
+    let commandBardata: ICommandBarData = {
+      primaryItems: [...items],
+      overflowItems: [...overflowItems!],
+      originalOverflowItems: [...overflowItems!], // for tracking
+      farItems,
+      cacheKey: '',
+    };
 
     return (
-      <div className={ css('ms-CommandBar', styles.root, className) } ref='commandBarRegion'>
-        { searchBox }
-        <FocusZone ref='focusZone' className={ styles.container } direction={ FocusZoneDirection.horizontal } role='menubar' >
-          <div className={ css('ms-CommandBar-primaryCommands', styles.primaryCommands) } ref='commandSurface'>
-            { renderedItems!.map((item, index) => (
-              this._renderItemInCommandBar(item, index, expandedMenuItemKey!)
-            )).concat((renderedOverflowItems && renderedOverflowItems.length) ? [
-              <div className={ css('ms-CommandBarItem', styles.item) } key={ OVERFLOW_KEY } ref={ OVERFLOW_KEY }>
-                <button
-                  type='button'
-                  id={ this._id + OVERFLOW_KEY }
-                  className={ css('ms-CommandBarItem-link', styles.itemLink, {
-                    ['is-expanded ' + styles.itemLinkIsExpanded]: (expandedMenuItemKey === OVERFLOW_KEY)
-                  }) }
-                  onClick={ this._onOverflowClick }
-                  role='menuitem'
-                  aria-label={ this.props.elipisisAriaLabel || '' }
-                  aria-haspopup={ true }
-                  data-automation-id='commandBarOverflow'
-                >
-                  <Icon className={ css('ms-CommandBarItem-overflow', styles.itemOverflow) } iconName='more' />
-                </button>
-              </div>
-            ] : []) }
-          </div>
-          <div className={ css('ms-CommandBar-sideCommands', styles.sideCommands) } ref='farCommandSurface'>
-            { renderedFarItems!.map((item, index) => (
-              this._renderItemInCommandBar(item, index, expandedMenuItemKey!, true)
-            )) }
-          </div>
-        </FocusZone>
-        { (contextualMenuProps) ?
-          (<ContextualMenu
-            className={ css('ms-CommandBar-menuHost') }
-            directionalHint={ DirectionalHint.bottomAutoEdge }
-            { ...contextualMenuProps }
-            targetElement={ contextualMenuTarget }
-            labelElementId={ expandedMenuId }
-            onDismiss={ this._onContextMenuDismiss }
-          />
-          ) : (null)
-        }
-      </div >
+      <ResizeGroup
```

Comment:
Is the measurement being cached accurate then? When rendering in the hidden div, won't it put a space between the elements?